### PR TITLE
release/v3.4.x: bump to SDK v0.45.10

### PR DESCRIPTION
**Pull request by faddat**
## Description

Improve the v3 series
---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
 
+retract [v3.0.0, v3.3.0]
+
 module github.com/cosmos/ibc-go/v3
 
+// dragonberry replace for ics23
+replace github.com/confio/ics23/go => github.com/cosmos/cosmos-sdk/ics23/go v0.8.0
+
+// protocol buffers replace
 replace github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
```
Comment (colin-axner) [inline go.mod:11]: nit: should we group the `replace`s at the bottom:

```go
replace (
    // ics23
    // protocol buffers
)

File: go.mod [hunk 2]
```diff
@@ -1,13 +1,19 @@
 go 1.18
 
+retract [v3.0.0, v3.3.0]
```
Comment (colin-axner) [inline go.mod:3]: should we add a comment? Froms go [docs](https://go.dev/ref/mod#go-mod-file-retract)

> Each retract directive should have a comment explaining the rationale for the retraction, though this is not mandatory. The go command may display rationale comments in warnings about retracted versions and in go list output. A rationale comment may be written immediately above a retract directive (without a blank line in between) or afterward on the same line. If a comment appears above a block, it applies to all retract directives within the block that don’t have their own comments. A rationale comment may span multiple lines.

Maybe something like "retracted version depends on SDK version without the dragonberry fix"

File: modules/apps/27-interchain-accounts/controller/keeper/account.go [hunk 3]
```diff
@@ -51,12 +52,14 @@ func (k Keeper) RegisterInterchainAccount(ctx sdk.Context, connectionID, owner s
...
 	)
 
-	versionBytes, err := icatypes.ModuleCdc.MarshalJSON(&metadata)
+	versionbytes, err := icatypes.ModuleCdc.MarshalJSON(&metadata)
 	if err != nil {
 		return err
 	}
+	version := string(versionbytes)
+
+	msg := channeltypes.NewMsgChannelOpenInit(portID, version, channeltypes.ORDERED, []string{connectionID}, icatypes.PortID, authtypes.NewModuleAddress(icatypes.ModuleName).String())
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/controller/keeper/account.go:62]: Changes in this file should be discarded

**Comment by colin-axner** [comment]
Hi @faddat, thanks for doing this work. Could you give us permissions to push to your branches so we could apply some of suggested changes? You can do so by clicking "Allow edits by maintainers". Do you have reservations with doing so?

**Comment by faddat** [comment]
Absolutely, re: access.  Please just let me know who to add to notional-labs/ibc-go and I'll do right away.  fixing conflicts and nits now.

File: go.mod (hunk 1 above)
Comment (faddat) [reply go.mod:11]: sure thing

File: go.mod (hunk 2 above)
Comment (faddat) [reply go.mod]: very yes

File: modules/apps/27-interchain-accounts/controller/keeper/account.go (hunk 3 above)
Comment (faddat) [reply modules/apps/27-interchain-accounts/controller/keeper/account.go]: ok

**Comment by crodriguezvega** [comment]
@faddat I opened [this PR](https://github.com/notional-labs/ibc-go/pull/105) to address some review comment. If you would be so kind to merge it, then we can merge this one.

**Comment by colin-axner** [comment]
> Absolutely, re: access. Please just let me know who to add to notional-labs/ibc-go 

Is there no "Allow edits by maintainers" on the right hand side of pr's? This might be the case, I have heard it can disappear based on your github settings. If not, could you add the code owners of this repo for write access?

**Comment by colin-axner** [comment]
Hi @faddat, we will take over these pr's. Thanks for your contributions :) I'm sure we will find ways to collaborate better in the future

**Review by colin-axner — APPROVED** [review]

**Review by crodriguezvega — APPROVED** [review]

**Pull request merged**